hnn_core/opt_toy_example/optimize_evoked.py

```diff
@@ -0,0 +1,162 @@
+            'data/MEG_detection_data/S1_SupraT.txt')
+urlretrieve(data_url, 'S1_SupraT.txt')
+exp_dpl = read_dipole('S1_SupraT.txt')
+
+###############################################################################
+# Now we start the optimization!
+#
+# First, we define a function that will tell the optimization routine how to
+# modify the network drive parameters. The function will take in the Network
+# object with no attached drives, and a dictionary of the paramters we wish to
+# optimize.
+
+
+def set_params(net, params_dict):
+
+    # Proximal 1
+    weights_ampa_p1 = {'L2_basket': params_dict['evprox1_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evprox1_ampa_L2_pyramidal'],
+                       'L5_basket': params_dict['evprox1_ampa_L5_basket'],
+                       'L5_pyramidal': params_dict['evprox1_ampa_L5_pyramidal']}
+    synaptic_delays_p = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                        'L5_basket': 1., 'L5_pyramidal': 1.}
+
+    net.add_evoked_drive('evprox1',
+                         mu=params_dict['evprox1_mu'],
+                         sigma=params_dict['evprox1_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p1,
+                         synaptic_delays=synaptic_delays_p)
+
+    # Distal
+    weights_ampa_d1 = {'L2_basket': params_dict['evdist1_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evdist1_ampa_L2_pyramidal'],
+                       'L5_pyramidal': params_dict['evdist1_ampa_L5_pyramidal']}
+    weights_nmda_d1 = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
+                       'L5_pyramidal': 0.080074}
+    synaptic_delays_d1 = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                          'L5_pyramidal': 0.1}
+
+    net.add_evoked_drive('evdist1',
+                         mu=params_dict['evdist1_mu'],
+                         sigma=params_dict['evdist1_sigma'],
+                         numspikes=1,
+                         location='distal',
+                         weights_ampa=weights_ampa_d1,
+                         weights_nmda=weights_nmda_d1,
+                         synaptic_delays=synaptic_delays_d1)
+
+    # Proximal 2
+    weights_ampa_p2 = {'L2_basket': params_dict['evprox2_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evprox2_ampa_L2_pyramidal'],
+                       'L5_basket': params_dict['evprox2_ampa_L5_basket'],
+                       'L5_pyramidal': params_dict['evprox2_ampa_L5_pyramidal']}
+
+    net.add_evoked_drive('evprox2',
+                         mu=params_dict['evprox2_mu'],
+                         sigma=params_dict['evprox2_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p2,
+                         synaptic_delays=synaptic_delays_p)
+
+    return net
```

Comment:
Is it actually necessary to `return net`?

These functions modify `net` in place, so if the copy occurs outside of this function then you just need to call the function on net